torch/utils/data/datapipes/iter/combining.py

```diff
@@ -223,6 +237,33 @@ def __len__(self):
     def get_generator_by_instance(self, instance_id: int):
         yield from self.main_datapipe.get_next_element_by_instance(self.instance_id)
 
+    # This method is called by `hook_iterator` in `_typing.py`.
+    def _set_main_datapipe_valid_iterator_id(self) -> int:
```

Comment:
Let me know if the explanation here is unclear. Also see the docstring/example within `class ChildDataPip`.